src/tests/java/picard/analysis/CollectGcBiasMetricsTest.java

```diff
@@ -0,0 +1,217 @@
+package picard.analysis;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMFileWriter;
+import htsjdk.samtools.SAMFileWriterFactory;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordSetBuilder;
+import htsjdk.samtools.SAMTextHeaderCodec;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.util.BufferedLineReader;
+import htsjdk.samtools.util.IOUtil;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+import picard.cmdline.CommandLineProgramTest;
+import picard.sam.SortSam;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Random;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Created by kbergin on 3/26/15 to test GcBias MultiLevel Collector.
+ */
+public class CollectGcBiasMetricsTest extends CommandLineProgramTest {
+    final static String sample1 = "TestSample1";
+    final static String sample2 = "TestSample2";
+    final static String readGroupId1 = "TestReadGroup1";
+    final static String readGroupId2 = "TestReadGroup2";
+    final static String readGroupId3 = "TestReadGroup3";
+    final static String platform = "ILLUMINA";
+    final static String library1 = "TestLibrary1";
+    final static String library2 = "TestLibrary2";
+    final static String library3 = "TestLibrary3";
+
+    final static File TEST_DIR = new File("testdata/picard/sam/CollectGcBiasMetrics/");
+    final File dict = new File(TEST_DIR, "Mheader.dict");
+
+    File tempDir;
+    File tempSamFile;
+    File samFile;
+
+    SAMRecordSetBuilder setBuilder1 = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+    SAMRecordSetBuilder setBuilder2 = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+    SAMRecordSetBuilder setBuilder3 = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+    SAMReadGroupRecord readGroupRecord1 = new SAMReadGroupRecord(readGroupId1);
+    SAMReadGroupRecord readGroupRecord2 = new SAMReadGroupRecord(readGroupId2);
+    SAMReadGroupRecord readGroupRecord3 = new SAMReadGroupRecord(readGroupId3);
+
+    //create a samfile with different samples, read groups and libraries that overlap for testing.
+    @BeforeTest
+    void setupBuilder() throws IOException {
+        final int numReads = 100;
+        final String flowCellBarcode = "TESTBARCODE";
+        final String readName = flowCellBarcode;
+
+        tempSamFile = File.createTempFile("CollectGcBias", ".bam", TEST_DIR);
+        File tempSamFileUnsorted = File.createTempFile("CollectGcBias", ".bam", TEST_DIR);
```

Comment:
Minor comment.  You can use outfile.deleteOnExit(); to set these up for deletion now - kind of nice.